src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,436 @@
+        if (current == address(0)) {
+            current = original;
+        }
+        require(current != to, "CURRENT_=_TO");
+        require(current == sender, "CURRENT_!=_SENDER");
+        if (to == original) {
+            _creatorship[original] = address(0);
+        } else {
+            _creatorship[original] = to;
+        }
+        emit CreatorshipTransfer(original, current, to);
+    }
+
+    /// @notice Set the Minter that will be the only address able to create Estate.
+    /// If set at deployment, resetting to address(0) will allow anyone to mint games
+    /// @param minter address of the minter
+    function setMinter(address minter) external override {
+        require(msg.sender == _admin, "ADMIN_NOT_AUTHORIZED");
+        require(minter != _minter, "MINTER_SAME_ALREADY_SET");
+        _minter = minter;
+        emit Minter(minter);
+    }
+
+    /// @notice Function to create a new GAME token
+    /// @param from The address of the one creating the game (may be different from msg.sender if metaTx)
+    /// @param to The address who will be assigned ownership of this game
+    /// @param assetIds The ids of the assets to add to this game
+    /// @param values the amount of each token id to add to game
+    /// @param editors The addresses to allow to edit (can also be set later)
+    /// @return id The id of the new GAME token (erc721)
+    function createGame(
+        address from,
+        address to,
+        uint256[] memory assetIds,
+        uint256[] memory values,
+        address[] memory editors,
+        string memory uri
+    ) external override minterGuard() notToZero(to) returns (uint256 id) {
+        // @review support metaTransactions ?
+        require(to != address(this), "DESTINATION_GAME_CONTRACT");
+        uint256 gameId = _mintGame(from, to);
+
+        if (editors.length != 0) {
+            for (uint256 i = 0; i < editors.length; i++) {
+                _gameEditors[gameId][editors[i]] = true;
+            }
+        }
+
+        if (assetIds.length != 0) {
+            if (assetIds.length == 1 && values[0] == 1) {
+                addSingleAsset(from, gameId, assetIds[0], uri);
+            } else {
+                addMultipleAssets(from, gameId, assetIds, values, uri);
+            }
+        }
+        setTokenURI(gameId, uri);
+        emit AssetsAdded(gameId, assetIds, values);
+        return gameId;
+    }
+
+    /// @notice Function to burn a GAME token
+    /// @param from The address of the one destroying the game
+    /// @param to The address to send all game assets to
+    /// @param gameId The id of the game to destroy
+    function destroyGame(
+        address from,
+        address to,
+        uint256 gameId
+    ) external override minterGuard() {
+        require(from == _ownerOf(gameId), "DESTROY_ACCESS_DENIED");
+        require(to != address(this), "DESTINATION_GAME_CONTRACT");
+        (uint256[] memory assets, uint256[] memory values) = getGameAssets(gameId);
```

Comment:
We need to consider the case where there is too many asset to remove in one block
See how Estate handle that by allowing the NFT to be burned but still support extraction of Land from it (by storing the previous owner)

Reply:
Fixed.